stdlib/public/core/Span/Span.swift

```diff
@@ -12,236 +12,270 @@
+extension Span where Element: BitwiseCopyable {
 
-  /// Unsafely create a `Span` over initialized memory.
+  /// Unsafely creates a `Span` over initialized memory.
   ///
-  /// The memory in `unsafeBytes` must be owned by the instance `owner`
-  /// meaning that as long as `owner` is alive the memory will remain valid.
+  /// The memory in `buffer` must remain valid throughout the lifetime of
+  /// the newly-created `Span`.
   ///
-  /// `unsafeBytes` must be correctly aligned for accessing
+  /// `buffer` must be correctly aligned for accessing
   /// an element of type `Element`, and must contain a number of bytes
   /// that is an exact multiple of `Element`'s stride.
   ///
   /// - Parameters:
-  ///   - unsafeBytes: a buffer to initialized elements.
+  ///   - buffer: a buffer to initialized elements.
   ///   - type: the type to use when interpreting the bytes in memory.
   ///   - owner: a binding whose lifetime must exceed that of
   ///            the newly created `Span`.
+  @_disallowFeatureSuppression(NonescapableTypes)
   @_alwaysEmitIntoClient
+  @lifetime(immortal)
   public init(
     _unsafeBytes buffer: UnsafeRawBufferPointer
-  ) -> dependsOn(immortal) Self {
+  ) {
+    _precondition(
+      ((Int(bitPattern: buffer.baseAddress) &
+        (MemoryLayout<Element>.alignment&-1)) == 0),
+      "baseAddress must be properly aligned to access Element"
+    )
     let (byteCount, stride) = (buffer.count, MemoryLayout<Element>.stride)
-    precondition(byteCount >= 0, "Count must not be negative")
     let (count, remainder) = byteCount.quotientAndRemainder(dividingBy: stride)
-    precondition(remainder == 0)
-    self.init(
-      _unchecked: buffer.baseAddress?.assumingMemoryBound(to: Element.self),
-      count: count
-    )
+    _precondition(remainder == 0, "Span must contain a whole number of elements")
+    self.init(_unchecked: buffer.baseAddress, count: count)
   }
 
-  /// Unsafely create a `Span` over initialized memory.
+  /// Unsafely creates a `Span` over initialized memory.
   ///
-  /// The memory in `unsafeBytes` must be owned by the instance `owner`
-  /// meaning that as long as `owner` is alive the memory will remain valid.
+  /// The memory in `buffer` must remain valid throughout the lifetime of
+  /// the newly-created `Span`.
   ///
-  /// `unsafeBytes` must be correctly aligned for accessing
+  /// `buffer` must be correctly aligned for accessing
   /// an element of type `Element`, and must contain a number of bytes
   /// that is an exact multiple of `Element`'s stride.
   ///
   /// - Parameters:
-  ///   - unsafeBytes: a buffer to initialized elements.
+  ///   - buffer: a buffer to initialized elements.
   ///   - type: the type to use when interpreting the bytes in memory.
   ///   - owner: a binding whose lifetime must exceed that of
   ///            the newly created `Span`.
+  @_disallowFeatureSuppression(NonescapableTypes)
   @_alwaysEmitIntoClient
+  @lifetime(immortal)
```

Comment:
Should be `@lifetime(buffer)`. The resulting Span depends on the validity of the buffer.